src/main/java/org/checkerframework/languageserver/CFDiagnostic.java

```diff
@@ -9,17 +9,37 @@
  * JSON wrapper to communicate javac diagnostics from {@link JavacWrapper} to {@link CheckExecutor}.
  */
 public class CFDiagnostic implements Diagnostic<Object> {
-
+    /** The file URI. */
     private final String fileUri;
+
+    /** The diagnostics kind. It is one of value in {@link Kind} enum. */
     private final String kind;
+
+    /**
+     * The diagnostics position. The position, startPosition, and endPosition satisfies the
+     * conditions getStartPosition() <= getPosition() <= getEndPosition().
+     */
     private final long position;
+
+    /** The diagnostics start position. */
     private final long startPosition;
+
+    /** The diagnostics end position. */
     private final long endPosition;
+
+    /** The line number. */
     private final long lineNumber;
+
+    /** The column number. */
     private final long columnNumber;
+
+    /** The code indicating the type of diagnostic. */
```

Comment:
Maybe an example would help here? In SourceChecker this is the `messageKey`?
https://github.com/eisop/checker-framework/blob/d7f9dabf1240e40484ce2b3c1c6e9feecaabe29c/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L1206

Reply:
yes, I have add dereference.of.nullable as an exmaple